ext/strings.go

```diff
@@ -83,7 +83,7 @@ import (
 //     'hello mellow'.lastIndexOf('')         // returns 12
 //     'hello mellow'.lastIndexOf('ello')     // returns 7
 //     'hello mellow'.lastIndexOf('jello')    // returns -1
-//     'hello mellow'.lastIndexOf('ello', 8)  // returns 2
+//     'hello mellow'.lastIndexOf('ello', 6)  // returns 2
```

Comment:
Shouldn't this return 1?

```
0 1 2 3 4 5 6 7 8 9
 h e l l o _ m e l l o
```

Also:

- specify that we return the index of the start of the matched string;
- the optional index is the _end_ of the search (assuming we're searching from left-to-right, as the name "`lastIndexOf`" implies.

Reply:
It should be `1`. The optional index represents where to start the substring match starting from the end, rather than where the substring match must end. 

Given input:
```
0 1 2 3 4 5 6 7 8 9 10
h e l l o _ m e l l o
```
Given an index of `7`, the substring match `ello` exactly starts at `7`.